src/pages/public/Login.tsx

```diff
@@ -1,24 +1,38 @@
 import { Form, Formik, FormikProps } from "formik";
+import * as queryString from "querystring";
 import React from "react";
+import { useDispatch } from "react-redux";
 import { Link } from "react-router-dom";
 import { object as yupObject, string as yupString } from "yup";
+import { login } from "~/actions/user";
 import { PrimaryButton } from "~/components/Button";
 import { InputField } from "~/components/Form";
+import { Title } from "~/components/Title";
 
 const initialValues = {
   email: "",
   password: ""
 };
 
 const validationSchema = yupObject().shape({
-  email: yupString().email("Ungültige E-Mail-Addresse").required("Pflichtfeld"),
+  email: yupString() /*.email("Ungültige E-Mail-Addresse")*/
+    .required("Pflichtfeld"),
   password: yupString().required("Pflichtfeld")
 });
 
-const Login = () => {
+const Login = ({ location }) => {
+  const dispatch = useDispatch();
+
+  const getRedirectRoute = () => {
+    let urlParams = queryString.parse(location && location.search);
+    return urlParams["?redirect"] || "/map";
+  };
```

Comment:
Spricht etwas dagegen, den [**URL constructor**](https://developer.mozilla.org/en-US/docs/Web/API/URL/URL) zu nutzen? Könnten uns so `querystring` als Import sparen.

Reply:
new URL ergibt ja nicht das gleiche oder ähnliche wie queryString.parse:

`For example, the query string 'foo=bar&abc=xyz&abc=123' is parsed into:

{
  foo: 'bar',
  abc: ['xyz', '123']
}`

Hier gehts ja darum, aus den URL params die Infos auszulesen, dazu hab ich bei URL() nichts gefunden.

Reply:
Ja, per `new URL()` erhält man Zugriff auf [**.searchParams**](https://developer.mozilla.org/en-US/docs/Web/API/URL/searchParams) und kann darüber auf einfache Weise mit Params interagieren. Sehe aber ein, dass es für diesen einfachen Check wie hier wohl Overkill ist. Habe in Node mal damit gearbeitet, weil man damit sehr zuverlässig params updaten kann etc. Zudem habe ich gesehen, dass die Lib sehr klein ist.

Können es daher dabei belassen. Reicht vollkommen aus.